### PR TITLE
Whole day x

### PR DESCRIPTION
**Pull request by austinjpaul**
Addresses #2 with some plugins. Hopefully it's not too ugly.
I also hard-coded the alerts timeframe, since traveltimes don't always line up (esp. with historical trip-id issues)

However.

This also introduces an error when switching out of multi-day mode into single-day mode.
Perhaps @mathcolo has some tricks to help?

### CODE REVIEW

**Comment by austinjpaul** [comment]
I forgot about the 2nd part of this: make the aggregate x always be the selected range, even when data is not available for some days. A similar approach should work, just have to work out the bugs.

File: src/line.js [hunk 1]
```diff
@@ -177,7 +177,11 @@ class LineClass extends React.Component {
                       display: true,
                       fontSize: 14,
                       labelString: this.props.xFieldLabel
-                    }
+                    },
+                    // make sure graph shows /at least/ suggestedXRange, either end will extend to not hide points.
+                    afterDataLimits: (axis) => {if (this.props.isLoading) return; // prevents weird sliding animation
+                                                axis.min = Math.min(axis.min, this.props.suggestedXRange[0]);
```
Comment (mathcolo) [inline src/line.js:183]: Sorry if I'm missing something, but can these statements just be gated to daily mode only?

File: src/App.js [hunk 2]
```diff
@@ -311,12 +312,37 @@ class App extends React.Component {
...
 
-  chartTimeframe() {
-    const travel_times = this.state.traveltimes;
-    if (travel_times.length > 0) {
-      return [new Date(travel_times[0].dep_dt), new Date(travel_times[travel_times.length - 1].dep_dt)];
+  suggestXRange() {
+    if (this.getTimescale() === 'hour') {
+      const today = `${this.state.configuration.date_start}T00:00:00`;
+
+      var low = new Date(today);
```
Comment (mathcolo) [inline src/App.js:319]: let 😃

File: src/App.js [hunk 3]
```diff
@@ -311,12 +312,37 @@ class App extends React.Component {
...
+  }
+
+  chartTimeframe() {
+    const today = `${this.state.configuration.date_start}T00:00:00`;
+
+    var low = new Date(today);
+    low.setHours(5, 30)
+
+    var high = new Date(today);
+    high.setDate(high.getDate() + 1);
```
Comment (mathcolo) [inline src/App.js:342]: This handles `32` and `31` correctly right?

File: src/line.js (hunk 1 above)
Comment (austinjpaul) [reply src/line.js]: I want to have this behavior for multi-day mode also
tmp

File: src/App.js (hunk 3 above)
Comment (austinjpaul) [reply src/App.js:342]: yep, it rolls over month and year.

**Comment by austinjpaul** [comment]
Bug fixed- this is working now!
tmp
tmp2

File: src/App.js (hunk 3 above)
Comment (austinjpaul) [reply src/App.js:342]: alternatively, i could just call setHours(25) ?

Comment (austinjpaul) [reply src/App.js:342]: lmk if you think something like this reads better: 
```      const { date_start } = this.state.configuration;

      const low = new Date(`${date_start}T06:00:00`);
      const high = new Date(`${date_start}T24:00:00`);
      high.setHours(1);

      return [low, high];
```

File: src/App.js [hunk 4]
```diff
@@ -311,12 +312,37 @@ class App extends React.Component {
...
 
-  chartTimeframe() {
-    const travel_times = this.state.traveltimes;
-    if (travel_times.length > 0) {
-      return [new Date(travel_times[0].dep_dt), new Date(travel_times[travel_times.length - 1].dep_dt)];
+  suggestXRange() {
+    if (this.getTimescale() === 'hour') {
+      const today = `${this.state.configuration.date_start}T00:00:00`;
+
+      let low = new Date(today);
```
Comment (mathcolo) [inline src/App.js:319]: Would you mind adding some comments to indicate that this is "6am today" and "1am tomorrow"?

File: src/App.js (hunk 3 above)
Comment (mathcolo) [reply src/App.js:342]: Hmmm no I think the way you have it is probably more clear with comments than this trick, but nice though lol.

**Review by mathcolo — APPROVED** [review]
Looks good!! Thanks for fixing this. A few comments, but otherwise I think good to merge.

**Pull request merged**